src/sentry/ratelimits/sliding_windows.py

```diff
@@ -142,7 +142,11 @@ def impl(self):
     def validate(self) -> None:
         try:
             self.client.ping()
-            self.client.connection_pool.disconnect()
+            if isinstance(self.client, RedisCluster):
+                self.client.disconnect_connection_pools()
+            else:
+                assert isinstance(self.client, StrictRedis)
+                self.client.connection_pool.disconnect()
```

Comment:
maybe a `def disconnect_redis_cluster(cluster: RedisCluster | StrictRedis) -> None: ...` ?

Reply:
makes sense, just implemented slight differently. thanks for the review @asottile-sentry